### ISSUE TITLE
Issue with sitemap: null

### ISSUE THREAD

**Issue by eDubrovsky** [mdreizin/gatsby-plugin-robots-txt#451]
Hi. After the plugin update, I'm getting an error when building my Gatsby page. In the configuration, I use ```sitemap: null```

Error:

``` ERROR #11321  PLUGIN

"gatsby-plugin-robots-txt" threw an error while running the onPostBuild lifecycle:

Invalid URL: null

  73 |   ) {
  74 |
> 75 |     mergedOptions.sitemap = new URL(path.posix.join(pathPrefix, 'sitemap', 'sitemap-index.xml'), mergedOptions.host).toString();
     |                             ^
  76 |   } else {
  77 |     try {
  78 |       new URL(mergedOptions.sitemap)

File: node_modules/gatsby-plugin-robots-txt/src/gatsby-node.js:75:29



  Error: TypeError [ERR_INVALID_URL]: Invalid URL: null
```

**Comment by holm** [comment]
I am seeing the same issue after the update. The plugin documentation explicitly recommends setting `sitemap` to null for dev/test.

**Comment by holm** [comment]
Seems to have come from the changes in https://github.com/mdreizin/gatsby-plugin-robots-txt/pull/441

**Comment by laurenskling** [comment]
yes, #441 breaks the possibility to have `sitemap: null` as stated in the documentation

**Comment by holm** [comment]
cc @moonmeister

**Comment by moonmeister** [comment]
Yup, we just need a null check and some tests probably. I'll do it if I find time.

**Comment by mdreizin** [comment]
@moonmeister @laurenskling @eDubrovsky I am really sorry, but I don't have a free time to fix that issue. I would appreciate if someone could look at it and fix.

**Comment by moonmeister** [comment]
@mdreizin or whomever...I'm trying to understand how you're getting an error thrown here. the lines 59-69 of this code should check for this null value and properly fill in the site url from the `siteUrl` metadata. So if this value is truly `null` then you must not be setting `siteUrl`. The whole point of setting the plugin config `sitemap` value to `null` is to fallback to that other value...if both are null we can provide a cleaner error but there's nothing we can do. Can you better explain what's happening or why you expect to happen? Links to your `gatsby-config.js` files could be helpful in fixing this bug. Thanks

**Comment by moonmeister** [comment]
Just realized those existing if statements aren't null checks, they're checking for existing properties, null or not. I'll look at this again and see what what I can figure out.

**Comment by wezter96** [comment]
Has this issue been resolved?
I seem to get the same issue.

**Comment by moonmeister** [comment]
I don't believe so.

**Comment by wezter96** [comment]
@moonmeister  could I just reroll to an older version of the plugin and then this issue wouldn't exist?

**Comment by moonmeister** [comment]
Sure but you'll have other issues that this update solved. I may have time to check this out soon. Feel free to contribute a fix too.

**Comment by laurenskling** [comment]
Can we please have a release of the fix? 😅

**Comment by mdreizin** [comment]
Please check the latest version.

**Comment by Dananz** [comment]
> Please check the latest version.

@mdreizin I don't think that the npm version is synced with the current repo version.

**Comment by laurenskling** [comment]
The merge isn't released into a release, would that be possible @mdreizin ?

**Comment by eDubrovsky** [comment]
@mdreizin this fix not merged into the 1.6.8 version. 😿

**Comment by CassieSloan** [comment]
Updated to most recent version of gatsby-plugin-robots-txt and our builds aren't running bc of the above. Any updates on when this will be fixed?

**Comment by iamskok** [comment]
@mdreizin Please re-open the issue. The fix wasn't merged in `1.6.8`  [CI output](https://github.com/mdreizin/gatsby-plugin-robots-txt/runs/3167195609?check_suite_focus=true#step:12:150)

`semantic-release` didn't generate a new version. I think it happened because PR didn't include a commit message with the word `fix`.
